pkg/kubelet/container/helpers.go

```diff
@@ -93,6 +96,16 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 // the running container with its desired spec.
 func HashContainer(container *v1.Container) uint64 {
 	hash := fnv.New32a()
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		cResources := container.Resources
+		cResourcesAllocated := container.ResourcesAllocated
+		container.Resources = v1.ResourceRequirements{}
+		container.ResourcesAllocated = nil
+		defer func() {
```

Comment:
why the defer here?

Reply:
I zero out Resources and ResourcesAllocated fields for computing the hash since these fields are mutable for Vertical Scaling, and we don't want the mutation to trigger a hash changed detection that syncpod uses to restart containers (restart is needed when image is updated for e.g)
defer restores the zeroed out fields back to their original values after the hash computation is done.